From 34d934612a49d0e69ca90052637f1051e37f4998 Mon Sep 17 00:00:00 2001
From: Andy Weidenbaum <atweiden@ioiojo.com>
Date: Mon, 21 Aug 2023 07:56:32 +0000
Subject: [PATCH] drop problematic zh-hant/X_keyboard_extension.html

```
This should take around a minute.
error parsing /usr/share/doc/arch-wiki-docs/html/zh-hant/X_keyboard_extension.html
Traceback (most recent call last):
  File "/builddir/arch-wiki-lite-2023.08.21/wiki_lite.py", line 185, in <module>
    main()
  File "/builddir/arch-wiki-lite-2023.08.21/wiki_lite.py", line 160, in main
    text = html_to_text(html)
           ^^^^^^^^^^^^^^^^^^
  File "/builddir/arch-wiki-lite-2023.08.21/wiki_lite.py", line 123, in html_to_text
    return remove_escapes(remove_tags(html))
                          ^^^^^^^^^^^^^^^^^
  File "/builddir/arch-wiki-lite-2023.08.21/wiki_lite.py", line 102, in remove_tags
    html = body.findall(html)[0]
           ^^^^^^^^^^^^^^^^^^^^^
  IndexError: list index out of range
```
---
 wiki_lite.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/wiki_lite.py b/wiki_lite.py
index 3d9b523..f54690e 100644
--- a/wiki_lite.py
+++ b/wiki_lite.py
@@ -123,11 +123,15 @@ def html_to_text(html):
     return remove_escapes(remove_tags(html))
 
 def all_html_paths(path):
+    exclude_path = '/usr/share/doc/arch-wiki-docs/html/zh-hant/X_keyboard_extension.html'
     for root,dirs,files in os.walk(path):
         for f in files:
             if not f.endswith('html'):
                 continue
-            yield os.path.join(root, f)
+            full_path = os.path.join(root, f)
+            if full_path == exclude_path:
+                continue
+            yield full_path
 
 def generate_toc():
     toc = {}
-- 
2.41.0

